db/blip_handler.go

```diff
@@ -1068,7 +1071,7 @@ func (bh *blipHandler) sendProveAttachment(sender *blip.Sender, docID, name, dig
 // For each attachment in the revision, makes sure it's in the database, asking the client to
 // upload it if necessary. This method blocks until all the attachments have been processed.
 func (bh *blipHandler) downloadOrVerifyAttachments(sender *blip.Sender, body Body, minRevpos int, docID string) error {
-	return bh.db.ForEachStubAttachment(body, minRevpos,
+	return bh.db.ForEachStubAttachment(body, minRevpos, docID,
```

Comment:
As discussed, as a defensive fallback for the case where we issue a proveAttachment request against SG (ISGR) for a 3.0 attachment, we should return a typed 'AttachmentNotFound' error.  We should check for that error here on line 1088, and fall back to getAttachment handling.

Reply:
Sure, this change has been committed.